### PR TITLE
setUpRandom

### LINKED ISSUE #95: create function to initialize RNG seed - must work both for matlab (including "old" versions) AND octave
I had this function somewhere else but I think this is not ideal

```matlab
function [] = setUpRand()
    % Set up the randomizers for uniform and normal distributions.
    % It is of great importance to do this before anything else!
    rand('state', sum(100, clock));
    randn('state', sum(100, clock));
end
```

@marcobarilari I think this is much more your area of expertise than mine. Can I assign you to this?

Having one function we can reuse across all projects would be super useful.

And also we actually need it for our localizers I think.

### PR DESCRIPTION
**Pull request by marcobarilari**
not many info online for a best practice, anyway I think that this two are the main source to understand what is happening:

(1) [Replace Discouraged Syntaxes of rand and randn](https://nl.mathworks.com/help/matlab/math/updating-your-random-number-generator-syntax.html)

(2) [ClockRandSeed](http://psychtoolbox.org/docs/ClockRandSeed)

as you see there is a func from PTB but I think that in matlab might be better to use `RandStream.setGlobalStream`. From what I grasp is the best solution.

fix #95

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Oh nice.

I had gone the lazy way in this PR #105 and was reusing Sam Schwartzkopf implementation.

As the file is lost in the middle of this PR that covers almost" all but the kitchen sink", you can see the file itself on one of my branch [here](https://github.com/cpp-lln-lab/CPP_PTB/blob/remi-motion_prf/src/utils/setupRand.m).

See maybe if there is anything you can take from there. 

You function looks better refactored but I think Sam's covers VERY old matlab cases too.

**Comment by marcobarilari** [comment]
uh nice as well yours, I missed that otherwise I would have took in account, let me then mix them up. 

Though, true that it covers also older matlab versions but it contradicts a bit what we check in our `initEnv` func. 

Question is, how old should be your matlab to use the old rand seeders + that old version is ok with the rest of the CPP_PTB/BIDS repo.

**Comment by Remi-Gau** [comment]
I think that having functions that support older matlab / octave is not a bad idea for something like this small function (that could be reused by others). 

I think that to make our life easier we say that we will not "support" versions older than X is more for things where we would have to modify a whole bunch of function of with some `if Ver > X; else` because of a new feature in Octave / Matlab.

Does that make sense?

File: src/utils/setUpRand.m [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    %
+    % Resets the seed of the random number generator. Will "adapt" depending on the matlab/octave
+    % version.
+    % It is of great importance to do this before anything else!
+    %
+    % For an alternative from PTB see `ClockRandSeed`
+
+    seed = sum(100 * clock);
+
+    if isOctave
```
Comment (Remi-Gau) [inline src/utils/setUpRand.m:15]: I think this `if` block can be deleted no?

if we are on octave the series of try catch will eventually hit the correct option line 36, no?

maybe we can just modify the message printed to the screen to say  :

disp('Using Octave or outdated matlab randomizer...');

**Review by Remi-Gau — CHANGES_REQUESTED** [review]

File: src/utils/setUpRand.m (hunk 1 above)
Comment (marcobarilari) [reply src/utils/setUpRand.m]: yes good point

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
@marcobarilari we did not add the call to setUpRand in initPTB, right?

**Comment by marcobarilari** [comment]
yes we did not, doing now

**Pull request merged**